parts/linux/cloud-init/artifacts/mariner/cse_install_mariner.sh

```diff
@@ -57,4 +57,29 @@ cleanUpGPUDrivers() {
     rm -Rf $GPU_DEST
 }
 
+disableSystemdResolvedCache() {
+    SERVICE_FILEPATH="/etc/systemd/system/resolv-uplink-override.service"
+    touch ${SERVICE_FILEPATH}
+    cat << EOF >> ${SERVICE_FILEPATH}
+[Unit]
+Description=Symlink /etc/resolv.conf to /run/systemd/resolve/resolv.conf
+After=systemd-networkd.service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/ln -sf /run/systemd/resolve/resolv.conf /etc/resolv.conf
+RemainAfterExit=no
+
+[Install]
+RequiredBy=network-online.target kubelet.service
+EOF
+
+    systemctlEnableAndStart resolv-uplink-override || exit $ERR_SYSTEMCTL_START_FAIL
+
+}
+
+disableSystemdIptables() {
+    systemctlDisableAndStop iptables || exit $ERR_DISBALE_IPTABLES
+}
```

Comment:
since this is called in VHD baking time only, can we move this to tools_install_mariner.sh? the payload to cloudinit has size limit so we want to minimize it as much as we can